datasync/downloader/queue.go

```diff
@@ -504,8 +505,13 @@ func (q *queue) reserveHeaders(p *peerConnection, count int, taskPool map[common
 
 		// If we're the first to request this task, initialise the result container
 		index := int(header.Number.Int64() - int64(q.resultOffset))
-		if index >= len(q.resultCache) || index < 0 {
-			common.Report("index allocation went beyond available resultCache space")
+		if index >= len(q.resultCache) {
+			logger.Warn("exceeded index of reserveHeaders", "header.Number.Uint64()", header.Number.Uint64())
```

Comment:
@KimKyungup When this logic can be executed?

Reply:
In general, this case can not occur without a logical bug.

Reply:
What kind of situation caused the logical bug in https://github.com/klaytn/klaytn/issues/435?

Reply:
That case is that the `index` is resultCache size or more case.
If request/receive process is faster than fetching the result, it can occur.
`resultCache` size is 8192.
The root cause is that there is no throttling logic in requesting.
We need to improve this later. But this case rarely occurs.